src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[ARRAY_MAX_SIZE];
+        this.values = (V[]) new Object[ARRAY_MAX_SIZE];
+        this.size = 0;
+    }
+
+    private int findIndexKey(K key) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] == null ? key == null : keys[i].equals(key)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findIndexKey(key);
+        if (index != -1) {
+            values[index] = value;
+        } else if (size < ARRAY_MAX_SIZE) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
```

Comment:
a small tweak for readability, regarding the max size check - would be perfect to check it and throw an exception if it is full, or you can ignore it for now, since it is not in the scope of this task
```suggestion
            return;
        }
        keys[size] = key;
        values[size] = value;
        size++;
```

Reply:
something like  try catch (ArrayIndexOutOfBoundsException e) ?

Reply:
no, we do not need to catch anything, our requirements say that ma size of storage is 10, so to ensure that we won't have more than 10 elements and to not get the raw `ArrayIndexOutOfBoundsException` we should avoid the situation when we get it by checking if size is 10 and we attempt to put another key/value pair - then we throw our own 
 custom `RuntimeException` with a message about max storage size